platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/projectmodule/BuildTreeLocalComponentProvider.java

```diff
@@ -31,7 +31,7 @@
 public interface BuildTreeLocalComponentProvider {
     /**
      * Get the local component for the project identified by {@code projectIdentifier}. The returned metadata will
-     * use foreign component identifiers for local components originating from builds different from {@code currentBuild}.
+     * use foreign component identifiers for local components originating from builds different from {@code currentBuildPath}.
      */
-    LocalComponentGraphResolveState getComponent(ProjectComponentIdentifier projectIdentifier, BuildIdentifier currentBuild);
+    LocalComponentGraphResolveState getComponent(ProjectComponentIdentifier projectIdentifier, Path currentBuildPath);
```

Comment:
🤔 Is there a reason that we couldn't keep using the build identifier?  It looks like it some places we have a build identifier to compare to, but we end up comparing it to the Path anyways?  e.g., [DefaultBuildTreeLocalComponentProvider.java](https://github.com/gradle/gradle/pull/27606/files#diff-c104a9292e964cff2737f92a34f971753c311eac983774aade0df2656fab78fa)

I think `buildIdentifier.equals(otherBuildIdentifier)` is the same as `buildIdentifier.getBuildPath().equals(otherBuildIdentifier.getBuildPath())`

Reply:
We could have continued to use the build identifier, but since we are now injecting the `DomainObjectContext`, into `DefaultLocalComponentRegistry`, which exposes the build as a `Path`, I felt we didn't also need to inject the `BuildIdentifier`